src/System.Reflection.MetadataLoadContext/src/FromCoreRt/System/DefaultBinder.cs

```diff
@@ -0,0 +1,612 @@
+        }
+
+        // Return any exact bindings that may exist. (This method is not defined on the
+        //  Binder and is used by RuntimeType.)
+        public static PropertyInfo ExactPropertyBinding(PropertyInfo[] match, Type returnType, Type[] types, ParameterModifier[] modifiers)
+        {
+            if (match == null)
+                throw new ArgumentNullException(nameof(match));
+
+            PropertyInfo bestMatch = null;
+            int typesLength = (types != null) ? types.Length : 0;
+            for (int i = 0; i < match.Length; i++)
+            {
+                ParameterInfo[] par = match[i].GetIndexParameters();
+                int j;
+                for (j = 0; j < typesLength; j++)
+                {
+                    Type pCls = par[j].ParameterType;
+
+                    // If the classes  exactly match continue
+                    if (pCls != types[j])
+                        break;
+                }
+                if (j < typesLength)
+                    continue;
+                if (returnType != null && returnType != match[i].PropertyType)
+                    continue;
+
+                if (bestMatch != null)
+                    throw new AmbiguousMatchException(SR.Arg_AmbiguousMatchException);
+
+                bestMatch = match[i];
+            }
+            return bestMatch;
+        }
+
+        private static int FindMostSpecific(ParameterInfo[] p1, int[] paramOrder1, Type paramArrayType1,
+                                            ParameterInfo[] p2, int[] paramOrder2, Type paramArrayType2,
+                                            Type[] types, object[] args)
+        {
+            // A method using params is always less specific than one not using params
+            if (paramArrayType1 != null && paramArrayType2 == null) return 2;
+            if (paramArrayType2 != null && paramArrayType1 == null) return 1;
+
+            // now either p1 and p2 both use params or neither does.
+
+            bool p1Less = false;
+            bool p2Less = false;
+
+            for (int i = 0; i < types.Length; i++)
+            {
+                if (args != null && args[i] == Type.Missing)
+                    continue;
+
+                Type c1, c2;
+
+                //  If a param array is present, then either
+                //      the user re-ordered the parameters in which case
+                //          the argument to the param array is either an array
+                //              in which case the params is conceptually ignored and so paramArrayType1 == null
+                //          or the argument to the param array is a single element
+                //              in which case paramOrder[i] == p1.Length - 1 for that element
+                //      or the user did not re-order the parameters in which case
+                //          the paramOrder array could contain indexes larger than p.Length - 1 (see VSW 577286)
```

Comment:
We generally don't include internal bug numbers..

Reply:
I'll remove the bug number. Note that this file and others were ported from CoreRt and there is still a discussion here around the best way to treat these files since some files have diverged and others have not. Also note that in this particular case, DefaultBinder.cs can also be [found ](https://github.com/dotnet/corefx/blob/master/src/Common/src/CoreLib/System/DefaultBinder.cs) in the current CoreFx repo (and with the same bug number), but including that will neccitate including many more files.